code/main/core/configurationModule.brs

```diff
@@ -21,33 +21,44 @@ function _adb_ConfigurationModule() as object
     module = _adb_AdobeObject("com.adobe.module.configuration")
     module.Append({
         _CONFIG_KEY: AdobeAEPSDKConstants().CONFIGURATION,
-        _MEDIA_CONFIG_KEY_PREFIX: "edgemedia.",
+        ' edge configuration
+        ' example: {"edge.configId":"1234567890", "edge.domain":"xyz.net"}
         _edge_configId: invalid,
         _edge_domain: invalid,
-        ' example config = {edge.configId:"1234567890", edge.domain:"xyz.net"
-        _media_configuration: {},
+        ' media configuration
+        ' example: {"edgemedia.channel":"channel_x", "edgemedia.playerName": "player_y", "edgemedia.appVersion": "1.0.0"}
+        _media_channel: invalid,
+        _media_playerName: invalid,
+        _media_appVersion: invalid,
 
         updateConfiguration: function(configuration as object) as void
-            configId = _adb_optStringFromMap(configuration, m._CONFIG_KEY.EDGE_CONFIG_ID)
-            domain = _adb_optStringFromMap(configuration, m._CONFIG_KEY.EDGE_DOMAIN)
 
+            ' update Edge configuration
+            configId = _adb_optStringFromMap(configuration, m._CONFIG_KEY.EDGE_CONFIG_ID)
             if not _adb_isEmptyOrInvalidString(configId)
                 m._edge_configId = configId
             end if
 
             '  example domain: company.data.adobedc.net
+            domain = _adb_optStringFromMap(configuration, m._CONFIG_KEY.EDGE_DOMAIN)
             regexPattern = CreateObject("roRegex", "^((?!-)[A-Za-z0-9-]+(?<!-)\.)+[A-Za-z]{2,6}$", "")
             if not _adb_isEmptyOrInvalidString(domain) and regexPattern.isMatch(domain)
                 m._edge_domain = domain
             end if
 
             ' update Media configuration
-            for each key in configuration.keys()
-                if key.startsWith(m._MEDIA_CONFIG_KEY_PREFIX)
-                    ' add/overwrite the media configuration
-                    m._media_configuration[key] = configuration[key]
-                end if
-            end for
+            channel = _adb_optStringFromMap(configuration, m._CONFIG_KEY.MEDIA_CHANNEL)
```

Comment:
Do you think we should add a method 
optValidStringFromMap which will return either non empty string or invalid, so we dont need to check the string one more time.

this:
```
playerName = _adb_optStringFromMap(configuration,
m._CONFIG_KEY.MEDIA_PLAYER_NAME)
 if not _adb_isEmptyOrInvalidString(playerName)
     m._media_playerName = playerName
 end if
```

becomes:
```
m._media_playerName = _adb_optValidStringFromMap(configuration, m._CONFIG_KEY.MEDIA_PLAYER_NAME) 
```

Reply:
No, it won't work anyway. For example, in the case of overwriting the existing configuration, we still need to validate the existence of the media keys.